lib/vtls/bearssl.c

```diff
@@ -100,26 +109,38 @@ static CURLcode load_cafile(const char *path, br_x509_trust_anchor **anchors,
   br_x509_trust_anchor *new_anchors;
   size_t new_anchors_len;
   br_x509_pkey *pkey;
-  FILE *fp;
-  unsigned char buf[BUFSIZ], *p;
+  FILE *fp = 0;
+  unsigned char buf[BUFSIZ];
+  const unsigned char *p;
   const char *name;
   size_t n, i, pushed;
 
-  fp = fopen(path, "rb");
-  if(!fp)
+  if(source->type != CAFILE_SOURCE_PATH && source->type != CAFILE_SOURCE_BLOB)
     return CURLE_SSL_CACERT_BADFILE;
 
+  if(source->type == CAFILE_SOURCE_PATH) {
+    fp = fopen(source->data, "rb");
+    if(!fp)
+      return CURLE_SSL_CACERT_BADFILE;
+  }
+
   ca.err = CURLE_OK;
   ca.in_cert = FALSE;
   ca.anchors = NULL;
   ca.anchors_len = 0;
   br_pem_decoder_init(&pc);
   br_pem_decoder_setdest(&pc, x509_push, &ca);
-  for(;;) {
-    n = fread(buf, 1, sizeof(buf), fp);
-    if(n == 0)
-      break;
-    p = buf;
+  do {
+    if(source->type == CAFILE_SOURCE_PATH) {
+      n = fread(buf, 1, sizeof(buf), fp);
+      p = buf;
```

Comment:
Assigning p to buf for an n of zero is a change from previous coding which doesn't seem right.

Reply:
Since buf is just a local array I think the only difference is a bit more instructions executed per function call. Fixed